rfcs/0021-tvmc_add_support_for_micro_targets.md

```diff
@@ -0,0 +1,461 @@
+
+positional arguments:
+  PROJECT_DIR           Project dir with a image built.
+
+optional arguments:
+  -h, --help            show this help message and exit
+
+platforms:
+  {zephyr,arduino,template}
+                        you must selected a platform from the list. You can pass '-h' for a selected platform to list its options.
+    zephyr              select Zephyr platform.
+    arduino             select Arduino platform.
+    template            select an adhoc template.
+
+$ tvmc micro flash /tmp/x200 zephyr
+usage: tvmc micro flash PROJECT_DIR zephyr [--list-options] -o OPTION=VALUE [OPTION=VALUE ...]
+tvmc micro flash PROJECT_DIR zephyr: error: the following arguments are required: -o
+
+$ tvmc micro flash /tmp/x200 zephyr --list-options
+usage: tvmc micro flash PROJECT_DIR zephyr [--list-options] -o OPTION=VALUE [OPTION=VALUE ...]
+
+optional arguments:
+  --list-options        show all options/values for selected platforms/template.
+  -o OPTION=VALUE [OPTION=VALUE ...]
+                        zephyr_board={mimxrt1050_evk, mps2_an521, nrf5340dk_nrf5340_cpuapp,
+                                      nucleo_f746zg, nucleo_l4r5zi, qemu_cortex_r5, qemu_riscv32,
+                                      qemu_riscv64, qemu_x86, stm32f746g_disco}
+                          name of the Zephyr board to build for. (required)
+
+$ tvmc micro flash /tmp/x200 zephyr -o zephyr_board=stm32f746g_disco
+$
+```
+
+## Running a model
+[running-a-model]: #running-a-model
+
+As shown above by using `tvmc micro` subcommands `create-project`, `build`, and
+`flash` it's possible to get a model flashed and ready to run on a micro target.
+
+Running a model is naturally common to all targets (not exclusive on MCUs), so
+commands and options for running a model on microcontrollers are accommodated
+under the existing `tvmc run` command. To select a micro target the device
+'micro' needs to be specified with `--device micro`. When it happens `PATH`
+argument can be used not to select a compiled model module file (like currently
+it happens on non-micro targets) but to specify the project directory used for
+flashing the model of interest:
+
+```bash
+
+$ tvmc run --device micro -h
+usage: tvmc run [-h] [--device {cpu,cuda,cl,metal,vulkan,rocm,micro}] [--fill-mode {zeros,ones,random}] [-i INPUTS] [-o OUTPUTS] [--print-time] [--print-top N] [--profile] [--repeat N] [--number N]
+                [--rpc-key RPC_KEY] [--rpc-tracker RPC_TRACKER]
+                PATH
+
+positional arguments:
+  PATH                  path to the compiled module file or to the project directory (micro devices only).
+
+optional arguments:
+  -h, --help            show this help message and exit
+  --device {cpu,cuda,cl,metal,vulkan,rocm,micro}
+                        target device to run the compiled module. Defaults to 'cpu'
+
+[...]
+```
+
+Once both `micro` device and project directory are specified `--list-options`
```

Comment:
would `--help` not do this? it's okay to go with this route, just curious

Reply:
I think that `--list-options` is ok, given that it makes clear to an end user that those options are dynamic options, implemented by the underlying template.

Reply:
@areusch in essence, and specifically to `tvmc run` command,  this is related to the issue of `parse_known_args()` "intercepting" `--help` (and `-h`) because `add_help = True` in the parsers everywhere. Hence that's related to your other comment (in https://github.com/apache/tvm/pull/9229#discussion_r746137979). I was able to resolve it for `tvmc micro` tho, so `--help` and `--list-options` are aliases and show the dynamic options. Thus if we could keep `--list-options` in the RFC it would work for both `run` and `micro` subcommands when listing the options. But on the other hand I'll experiment as you suggested anyway (setting add_help=False on every parser)